media-plugins/adlplug/adlplug-9999.ebuild

```diff
@@ -0,0 +1,126 @@
+# Copyright 2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit cmake-utils multibuild git-r3
+
+DESCRIPTION="FM synthesizer plugins based on OPL3 and OPN2 sound chip emulations"
+HOMEPAGE="https://github.com/jpcima/ADLplug"
+
+EGIT_REPO_URI="https://github.com/jpcima/${PN}.git"
+if [[ ${PV} == *9999 ]]; then
+	KEYWORDS=""
+else
+	# Upstream release tarballs do not contain the necessary third party libs;
+	# these are available via git submodules per upstream build instructions
+	EGIT_COMMIT="v${PV}"
+	KEYWORDS="~amd64"
+fi
+
+LICENSE="Boost-1.0 BSD GPL-3 GPL-2+ LGPL-3 LGPL-2.1 MIT ISC"
+SLOT="0"
+
+# Upstream offers a VST3 build option, but unfortunately JUCE 5.4.1
+# doesn't support VST3 on Linux
+# see https://github.com/WeAreROLI/JUCE/issues/307
+IUSE="+opl3 +opn2 +lv2 +vst jack alsa pch assertions"
+REQUIRED_USE="
+	|| ( opl3 opn2 )
```

Comment:
The reason I wrote the ebuild this way is because of how the upstream build process works. There's no way to build both plugins in one go; you have to run cmake twice, changing the value of a parameter for each one (see [here](https://github.com/jpcima/ADLplug/blame/master/README.md#L73)).

As long as the build process has to run twice anyway, I figured we might as well give users the option to build only one of them if someone wants to. I think you're right that most users would always build both, which is why I made that the default.

I'd much rather have them in one package then split into two. Arch used to have them in two separate packages and I remember I found it cumbersome. It looks like they've made them into a split package now (which basically does the same thing this ebuild does; they share a single PKGBUILD).